pkg/ccl/testccl/sqlccl/sqlccl_test: TestCancelMultipleSessions failed

pkg/ccl/testccl/sqlccl/sqlccl_test.TestCancelMultipleSessions [failed](https://mesolite.cluster.engflow.com/invocations/default/f43ec2ae-fbe3-4f04-a9c7-ac0b9399d80a?testReportRun=1&testReportShard=7&testReportAttempt=1#targets-Ly9wa2cvY2NsL3Rlc3RjY2wvc3FsY2NsOnNxbGNjbF90ZXN0) on master @ [420d7da30c063edd781ab8eedbcafc78f1477847](https://github.com/cockroachdb/cockroach/commits/420d7da30c063edd781ab8eedbcafc78f1477847):


```
=== RUN   TestCancelMultipleSessions
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestCancelMultipleSessions3376786523
    test_log_scope.go:81: use -show-logs to present logs inline
    run_control_test.go:277: -- test log scope end --
--- FAIL: TestCancelMultipleSessions (3.05s)
=== RUN   TestCancelMultipleSessions/Tenant
    run_control_test.go:272: session 0 not canceled; actual error: read tcp 127.0.0.1:55488->127.0.0.1:36967: read: connection reset by peer
    --- FAIL: TestCancelMultipleSessions/Tenant (0.02s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=7</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCancelMultipleSessions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-38904

pkg/ccl/testccl/sqlccl/sqlccl_test.TestCancelMultipleSessions [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15339043?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15339043?buildTab=artifacts#/) on master @ [420d7da30c063edd781ab8eedbcafc78f1477847](https://github.com/cockroachdb/cockroach/commits/420d7da30c063edd781ab8eedbcafc78f1477847):


```
=== RUN   TestCancelMultipleSessions
I240520 23:27:34.992628 1 (gostd) rand.go:250  [-] 1  random seed: -8958428060124471480
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/d314124205517e8a124be605d4c14d7b/logTestCancelMultipleSessions3814858060
    test_log_scope.go:81: use -show-logs to present logs inline
    run_control_test.go:277: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/d314124205517e8a124be605d4c14d7b/logTestCancelMultipleSessions3814858060
--- FAIL: TestCancelMultipleSessions (1.98s)
=== RUN   TestCancelMultipleSessions/Tenant
    run_control_test.go:272: session 0 not canceled; actual error: read tcp 127.0.0.1:48016->127.0.0.1:35181: read: connection reset by peer
    --- FAIL: TestCancelMultipleSessions/Tenant (0.13s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCancelMultipleSessions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

pkg/ccl/testccl/sqlccl/sqlccl_test.TestCancelMultipleSessions [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15339656?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15339656?buildTab=artifacts#/) on master @ [a7aad51345cb598b5144d0f0f3028e55a121a6e1](https://github.com/cockroachdb/cockroach/commits/a7aad51345cb598b5144d0f0f3028e55a121a6e1):


```
=== RUN   TestCancelMultipleSessions
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/b88597f635494aa272c70f7c409cafd6/logTestCancelMultipleSessions3467342432
    test_log_scope.go:81: use -show-logs to present logs inline
    run_control_test.go:277: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/b88597f635494aa272c70f7c409cafd6/logTestCancelMultipleSessions3467342432
--- FAIL: TestCancelMultipleSessions (1.33s)
=== RUN   TestCancelMultipleSessions/Tenant
    run_control_test.go:272: session 0 not canceled; actual error: read tcp 127.0.0.1:51514->127.0.0.1:44949: read: connection reset by peer
    --- FAIL: TestCancelMultipleSessions/Tenant (0.03s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCancelMultipleSessions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

pkg/ccl/testccl/sqlccl/sqlccl_test.TestCancelMultipleSessions [failed](https://tanzanite.cluster.engflow.com/invocations/default/22d7d37f-9dfb-4e55-8b11-dffe4e6c749b?testReportRun=7&testReportShard=8&testReportAttempt=1#targets-Ly9wa2cvY2NsL3Rlc3RjY2wvc3FsY2NsOnNxbGNjbF90ZXN0) on master @ [d146ecff6f687e438706cf63591cafca60cc116d](https://github.com/cockroachdb/cockroach/commits/d146ecff6f687e438706cf63591cafca60cc116d):


```
=== RUN   TestCancelMultipleSessions
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestCancelMultipleSessions979438502
    test_log_scope.go:81: use -show-logs to present logs inline
    run_control_test.go:277: -- test log scope end --
test logs left over in: outputs.zip/logTestCancelMultipleSessions979438502
--- FAIL: TestCancelMultipleSessions (1.55s)
=== RUN   TestCancelMultipleSessions/Tenant
    run_control_test.go:272: session 0 not canceled; actual error: read tcp 127.0.0.1:35896->127.0.0.1:36375: read: connection reset by peer
    --- FAIL: TestCancelMultipleSessions/Tenant (0.02s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=7</code>
 - <code>shard=8</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCancelMultipleSessions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

pkg/ccl/testccl/sqlccl/sqlccl_test.TestCancelMultipleSessions [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15350949?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15350949?buildTab=artifacts#/) on master @ [081f60c7b6e57d2cf2e9346b4f65a21efb967513](https://github.com/cockroachdb/cockroach/commits/081f60c7b6e57d2cf2e9346b4f65a21efb967513):


```
=== RUN   TestCancelMultipleSessions
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/b88597f635494aa272c70f7c409cafd6/logTestCancelMultipleSessions3447255300
    test_log_scope.go:81: use -show-logs to present logs inline
    run_control_test.go:277: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/b88597f635494aa272c70f7c409cafd6/logTestCancelMultipleSessions3447255300
--- FAIL: TestCancelMultipleSessions (1.61s)
=== RUN   TestCancelMultipleSessions/Tenant
    run_control_test.go:272: session 0 not canceled; actual error: read tcp 127.0.0.1:49726->127.0.0.1:46149: read: connection reset by peer
    --- FAIL: TestCancelMultipleSessions/Tenant (0.03s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCancelMultipleSessions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

pkg/ccl/testccl/sqlccl/sqlccl_test.TestCancelMultipleSessions [failed](https://tanzanite.cluster.engflow.com/invocations/default/457a38e6-f105-4c19-9d3c-35e7c71abdd6?testReportRun=8&testReportShard=8&testReportAttempt=1#targets-Ly9wa2cvY2NsL3Rlc3RjY2wvc3FsY2NsOnNxbGNjbF90ZXN0) on master @ [8156d095961d7271df61fe2e2f716ef072bc5f96](https://github.com/cockroachdb/cockroach/commits/8156d095961d7271df61fe2e2f716ef072bc5f96):


```
=== RUN   TestCancelMultipleSessions
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestCancelMultipleSessions2593034577
    test_log_scope.go:81: use -show-logs to present logs inline
    run_control_test.go:277: -- test log scope end --
test logs left over in: outputs.zip/logTestCancelMultipleSessions2593034577
--- FAIL: TestCancelMultipleSessions (1.50s)
=== RUN   TestCancelMultipleSessions/Tenant
    run_control_test.go:272: session 1 not canceled; actual error: read tcp 127.0.0.1:54146->127.0.0.1:39965: read: connection reset by peer
    --- FAIL: TestCancelMultipleSessions/Tenant (0.02s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=8</code>
 - <code>shard=8</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCancelMultipleSessions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

pkg/ccl/testccl/sqlccl/sqlccl_test.TestCancelMultipleSessions [failed](https://tanzanite.cluster.engflow.com/invocations/default/fcd81239-bbee-4de5-bdcd-9184c8218783?testReportRun=1&testReportShard=8&testReportAttempt=1#targets-Ly9wa2cvY2NsL3Rlc3RjY2wvc3FsY2NsOnNxbGNjbF90ZXN0) on master @ [8156d095961d7271df61fe2e2f716ef072bc5f96](https://github.com/cockroachdb/cockroach/commits/8156d095961d7271df61fe2e2f716ef072bc5f96):


```
=== RUN   TestCancelMultipleSessions
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestCancelMultipleSessions975955164
    test_log_scope.go:81: use -show-logs to present logs inline
    run_control_test.go:277: -- test log scope end --
test logs left over in: outputs.zip/logTestCancelMultipleSessions975955164
--- FAIL: TestCancelMultipleSessions (13.75s)
=== RUN   TestCancelMultipleSessions/Tenant
    run_control_test.go:272: session 0 not canceled; actual error: read tcp 127.0.0.1:53318->127.0.0.1:42505: read: connection reset by peer
    --- FAIL: TestCancelMultipleSessions/Tenant (0.14s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=8</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCancelMultipleSessions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/ccl/testccl/sqlccl/run_control_test.go
+++ b/pkg/ccl/testccl/sqlccl/run_control_test.go
@@ -18,6 +18,7 @@ import (
 
 	"github.com/cockroachdb/cockroach/pkg/base"
 	"github.com/cockroachdb/cockroach/pkg/sql/sqltestutils"
+	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
@@ -220,7 +221,7 @@ func testCancelSession(t *testing.T, hasActiveSession bool) {
 				_, err = conn1.ExecContext(ctx, "SELECT 1")
 			}
 
-			if !errors.Is(err, gosqldriver.ErrBadConn) {
+			if !errors.Is(err, gosqldriver.ErrBadConn) && !testutils.IsError(err, "connection reset by peer") {
 				t.Fatalf("session not canceled; actual error: %s", err)
 			}
 		})
@@ -268,8 +269,8 @@ func TestCancelMultipleSessions(t *testing.T) {
 			// Verify that the connections on node 1 are closed.
 			for i := 0; i < 2; i++ {
 				_, err := conns[i].ExecContext(ctx, "SELECT 1")
-				if !errors.Is(err, gosqldriver.ErrBadConn) {
-					t.Fatalf("session %d not canceled; actual error: %s", i, err)
+				if !errors.Is(err, gosqldriver.ErrBadConn) && !testutils.IsError(err, "connection reset by peer") {
+					t.Fatalf("session %d not canceled; actual error: %v", i, err)
 				}
 			}
 		})

